packages/blade/src/components/Link/BaseLink/StyledBaseLink.web.tsx

```diff
@@ -1,33 +1,50 @@
+import React from 'react';
 import styled from 'styled-components';
 import getStyledLinkStyles from './getStyledLinkStyles';
 import type { StyledBaseLinkProps } from './types';
-import { castWebType, getIn, makeBorderSize, makeMotionTime } from '~utils';
+import { castWebType, getIn, getStyledProps, makeBorderSize, makeMotionTime } from '~utils';
+import { getBaseBoxStyles, getDependencyProps } from '~components/Box/BaseBox/getBaseBoxStyles';
 
 const StyledLink = styled.button.attrs<StyledBaseLinkProps>((props: StyledBaseLinkProps) => ({
   ...props.accessibilityProps,
-}))<StyledBaseLinkProps>((props) => ({
-  ...getStyledLinkStyles(props),
-  display: 'inline-block',
-  borderRadius: makeBorderSize(props.theme.border.radius.small),
-  transitionProperty: 'box-shadow',
-  transitionTimingFunction: getIn(props.theme.motion, props.motionEasing),
-  transitionDuration: castWebType(makeMotionTime(getIn(props.theme.motion, props.motionDuration))),
-  '.content-container': {
-    width: 'max-content',
+}))<StyledBaseLinkProps>((props) => {
+  // In other components, we wrap the component by div. But for link and text, we wanted to avoid that to not add unneccessary div around and cause layout any issues
+  // So we build the styled-props specifically for link and text
+  const styledPropsStyles = getStyledProps(props);
+  const styledPropsMemoDependency = getDependencyProps(styledPropsStyles);
+  const styledPropsCSSObject = React.useMemo(
+    () => getBaseBoxStyles({ ...styledPropsStyles, theme: props.theme }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [styledPropsMemoDependency],
+  );
+
+  return {
+    ...styledPropsCSSObject,
+    ...getStyledLinkStyles(props),
+    display: 'inline-block',
     borderRadius: makeBorderSize(props.theme.border.radius.small),
```

Comment:
Important part:

For `BaseLink` and `BaseText`, our approach is a bit different to add style-props. Instead of wrapping these components in BaseBox, I am just injecting all the responsive props from box into its css.

I tried wrapping them in BaseBox but they cause several issues in layouts and add some weird spacing sometimes. Overall too many things were breaking with that approach so I avoided adding new unnecessary wrapper.

Reply:
Seems a bit confusing especially because there's so much abstraction here. Can we discuss this once with the team?

Reply:
yeah sure. The reason this is so much of code is because we can't write `React.useMemo` inside one of those `get* functions. Let me create a separate hook and wrap in it that should be simpler to understand and add.

Reply:
created separate hook